hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -380,6 +386,61 @@ else if ( enhancementContext.doExtendedEnhancement( managedCtClass ) ) {
 		}
 	}
 
+	// See HHH-16572
+	// return true if enhancement is unsupported
+	private boolean unsupportedEnhancement(TypeDescription managedCtClass) {
+		boolean result = false;
+		// Check for use of ID/AccessType(PROPERTY) on methods
+
+		for (MethodDescription.InDefinedShape shape : managedCtClass.getDeclaredMethods()) {
```

Comment:
@raphw Does this use of getDeclaredMethods look correct to you?  

I'm trying to check the passed application class to see if it has any methods that have @jakarta.persistence.Id or @jakarta.persistence.Access (with a certain value) which seems to work with some testing but I'm seeing some Hibernate test failures that I do not yet understand.  This is to avoid rewriting classes that have those annotations to avoid the https://hibernate.atlassian.net/browse/HHH-16572 problem.  

Thank you for advice! 

Scott

Reply:
Are you only interested in the declared methods of a class, or also in the ones that are inherited from super classes. For the former, this is correct..For the latter, you'd need to use a MethodGraph.Compiler and resolve the hierarchy.

Reply:
@raphw I'm interested in the declared methods of a class and also the ones that are inherited from super classes.  I'm also interested in examining the declared fields (methodDescription.getDeclaringType().getDeclaredFields()) of each `methodDescription.getDeclaringType()`.

Is it correct for me to depend on `methodDescription.getDeclaringType().getDeclaredFields()` to access each class set of fields in the hierarchy?

```
		MethodGraph.Linked methodGraph = MethodGraph.Compiler.Default.forJavaHierarchy().compile(managedCtClass);
		for (MethodGraph.Node node : methodGraph.listNodes()) {
			MethodDescription methodDescription = node.getRepresentative();
			ArrayList<AnnotatedFieldDescription> fieldList = new ArrayList<>();
			for (FieldDescription ctField : methodDescription.getDeclaringType().getDeclaredFields()) {
					AnnotatedFieldDescription annotatedField = new AnnotatedFieldDescription(enhancementContext, ctField);
					if (enhancementContext.isPersistentField(annotatedField)) {
						fieldList.add(annotatedField);
					}
				}
			}
                        ...  validate each declared method name against the fields in methodDescription.getDeclaringType()



```

Reply:
As far as I can tell ^ code is fine except I should instead be looping through each class fields and validate those against the class declared methods.

Reply:
Or would it be better to use net.bytebuddy.dynamic.scaffold.FieldLocator to walk through all of the fields of each class in the hierarchy?

Reply:
A `FieldLocator` would resolve a field by its name as the Java language would do it, so this sounds like the right approach to me.

Reply:
I removed the unneeded checks that duplicated checks done in other code and stuck with the MethodGraph.Compiler as that covers the needs for the added check done by this change.